### PR TITLE
Merge decaffinated master into rebrand

### PR DESCRIPTION
**Pull request by Spiker985**
Because this branch was originally cherry-picked and it's last commit was 0fdf1ed, it was missing quite a few files from master.

The only merge conflict was `src/context-menu-manager.coffee` which was deleted from master, and originally had an A[a]tom -> P[p]ulsar rename within one of the comments.

All other files were auto-merged together.

Confirmed `yarn install && yarn build && yarn start` did function, and load the editor.

### CODE REVIEW

File: packages/language-gfm/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1 @@
+See the [Atom contributing guide](https://github.com/atom/atom/blob/master/CONTRIBUTING.md)
```
Comment (confused-Techie) [inline packages/language-gfm/CONTRIBUTING.md:1]:
```suggestion
See the [Pulsar contributing guide](https://github.com/pulsar-edit/.github/blob/main/CONTRIBUTING.md)
```

File: packages/language-gfm/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<!--
```
Comment (confused-Techie) [inline packages/language-gfm/ISSUE_TEMPLATE.md:1]: [Mirror?](https://github.com/pulsar-edit/.github/blob/main/.github/ISSUE_TEMPLATE/bug_report.md)

File: packages/language-gfm/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# GitHub flavored Markdown package
+[![OS X Build Status](https://travis-ci.org/atom/language-gfm.svg?branch=master)](https://travis-ci.org/atom/language-gfm) [![Windows Build Status](https://ci.appveyor.com/api/projects/status/rpub8qjyd8lt7wai/branch/master?svg=true)](https://ci.appveyor.com/project/Atom/language-gfm/branch/master) [![Dependency Status](https://david-dm.org/atom/language-gfm.svg)](https://david-dm.org/atom/language-gfm)
+
+Adds syntax highlighting and snippets to [GitHub flavored Markdown](https://help.github.com/articles/github-flavored-markdown) files in Atom.
```
Comment (confused-Techie) [inline packages/language-gfm/README.md:4]:
```suggestion
Adds syntax highlighting and snippets to [GitHub flavored Markdown](https://help.github.com/articles/github-flavored-markdown) files in Pulsar.
```

File: packages/language-gfm/package.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "language-gfm",
+  "version": "0.90.8",
+  "description": "Syntax highlighting and snippets for GitHub Flavored Markdown (GFM).",
+  "repository": "https://github.com/atom/language-gfm",
```
Comment (confused-Techie) [inline packages/language-gfm/package.json:5]:
```suggestion
  "repository": "https://github.com/pulsar/language-gfm",
```

File: packages/language-git/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1 @@
+See the [Atom contributing guide](https://github.com/atom/atom/blob/master/CONTRIBUTING.md)
```
Comment (confused-Techie) [inline packages/language-git/CONTRIBUTING.md:1]:
```suggestion
See the [Pulsar contributing guide](https://github.com/pulsar-edit/.github/blob/main/CONTRIBUTING.md)
```

File: packages/language-git/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+<!--
```
Comment (confused-Techie) [inline packages/language-git/ISSUE_TEMPLATE.md:1]: [Mirror?](https://github.com/pulsar-edit/.github/blob/main/.github/ISSUE_TEMPLATE/bug_report.md)

File: packages/language-git/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# Git editing support in Atom
```
Comment (confused-Techie) [inline packages/language-git/README.md:1]:
```suggestion
# Git editing support in Pulsar
```

File: packages/language-git/README.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+[![macOS Build Status](https://travis-ci.org/atom/language-git.svg?branch=master)](https://travis-ci.org/atom/language-git)
+[![Windows Build Status](https://ci.appveyor.com/api/projects/status/481319gyrr1feo8b/branch/master?svg=true)](https://ci.appveyor.com/project/Atom/language-git/branch/master)
+[![Dependency Status](https://david-dm.org/atom/language-git.svg)](https://david-dm.org/atom/language-git)
+
+Adds syntax highlighting to Git commit, merge, and rebase messages edited in Atom.
```
Comment (confused-Techie) [inline packages/language-git/README.md:6]:
```suggestion
Adds syntax highlighting to Git commit, merge, and rebase messages edited in Pulsar.
```

File: packages/language-git/README.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+You can configure Atom to be your Git editor with the following command:
```
Comment (confused-Techie) [inline packages/language-git/README.md:8]:
```suggestion
You can configure Pulsar to be your Git editor with the following command:
```

File: package.json [hunk 10]
```diff
@@ -36,7 +36,7 @@
     "atom-select-list": "^0.8.1",
```
[the remark this answers is not in the document]
Comment (Spiker985) [reply package.json:36]: At this juncture I was going to leave all existing packages, and internal references alone, while still replacing the moniker at necessarily non-key intersections.

Primarily just to pare down the count of references

**Comment by confused-Techie** [comment]
Since speaking with @Spiker985 previously he is fine with me going ahead and merging recommendations on the Markdown files themselves, so I will clear those out of my requested changes. Otherwise as its been open for nearly a week, and there doesn't seem to be any other issues found how do we feel about merging into main @pulsar-edit/core @pulsar-edit/documentation

**Comment by confused-Techie** [comment]
How do we feel about merging some JSON data rebranding into this PR, then commit to main? If anyone else wants to go ahead and approve it we could probably commit afterwards

**Comment by confused-Techie** [comment]
I'm gonna go ahead and merge this into the rebrand branch for `pulsar-edit/pulsar`. So that does mean this will afterwards have to be merged to main.

But its been up for some time, and while I think there can still be advantages of modifying the JSON and some other aspects I will leave those up to another PR as this one is already large enough.

Otherwise as for testing our expected failure rate of 55 Editor Tests and 22 Package Tests this does fall short on some of those. But it looks like this PR was created before those goals were set or expected.

Its current stats look like so:

* ubuntu
  * Editor: 63 
  * Package: 22
* mac
  * Editor: Truncated
  * Package: 22
* windows
  * Editor: 38
  * Package: Failed to Install

So while on Ubuntu the Editor experiences more errors than we like, on Windows its the exact opposite of what we would expect, so I will merge in hopes these can be ironed out before merging to main.

**Pull request merged**